addons/account/models/account_payment.py

```diff
@@ -659,6 +663,24 @@ def _compute_payment_receipt_title(self):
         """ To override in order to change the title displayed on the payment receipt report """
         self.payment_receipt_title = _('Payment Receipt')
 
+    @api.depends('partner_id', 'amount', 'date')
+    def _compute_duplicated_payment_ids(self):
+        """ Retrieve ids of payments with same partner_id, amount and date as the current payment """
+        for payment in self:
+            payment.has_duplicated_payments = 0
+            payment.duplicated_payment_ids = None
+            if payment.partner_id.id and payment.amount != 0:
+                domain = [('partner_id', '=', payment.partner_id.id),
+                          ('amount', '=', payment.amount),
+                          ('date', '=', payment.date),
+                          ('state', 'in', ('posted', 'draft')),]
+                if not isinstance(payment.id, models.NewId):
+                    domain.append(('id', '!=', payment.id))
+                duplicates = self.env['account.payment'].search(domain)
```

Comment:
We usually try to avoid using search in a loop for performance issues, i think you will have to do a thing similar duplicated_ref_ids for moves

Reply:
Added the fetch duplicate reference function instead